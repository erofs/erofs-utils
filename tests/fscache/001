#!/bin/sh
# SPDX-License-Identifier: GPL-2.0+
#
# check inline/non-inline/multidev layout in fscache mode
#
seq=`basename $0`
seqres=$RESULT_DIR/$(echo $0 | awk '{print $((NF-1))"/"$NF}' FS="/")

# get standard environment, filters and checks
. "${srcdir}/common/rc"

cleanup()
{
	pkill cachefilesd2
	# wait for cachefiles daemon exiting, or the backing files
	# can't be deleted since they are still IN_USE.
	sleep 1

	cd /
	rm -rf $tmp.*
}

_require_erofs
_require_fscache

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
echo "QA output created by $seq"

if [ -z $SCRATCH_DEV ]; then
	SCRATCH_DEV=$tmp/erofs_$seq.img
	rm -f SCRATCH_DEV
fi

localdir="$tmp/$seq"
rm -rf $localdir
mkdir -p $localdir

# collect files
echo "test" > $localdir/testfile

# start cachefilesd daemon
($CACHEFILESD_PROG $tmp $tmp >> $seqres.full 2>&1 &)

# test inline layout
echo "[INLINE]" >> $seqres.full
_scratch_mkfs $localdir >> $seqres.full 2>&1 || _fail "failed to mkfs"
SCRATCH_OPTIONS="-o fsid=erofs_${seq}.img"
_scratch_mount 2>>$seqres.full
cat $SCRATCH_MNT/testfile
_scratch_unmount

# test noni-inline layout
echo "[NON_INLINE]" >> $seqres.full
_scratch_mkfs $localdir -Enoinline_data >> $seqres.full 2>&1 || _fail "failed to mkfs"
SCRATCH_OPTIONS="-o fsid=erofs_${seq}.img"
_scratch_mount 2>>$seqres.full
cat $SCRATCH_MNT/testfile
_scratch_unmount

# test multidev layout
echo "[MULTIDEV]" >> $seqres.full
_scratch_mkfs $localdir --chunksize=1048576 --blobdev=$tmp/blob.img -Eforce-chunk-indexes \
	>> $seqres.full 2>&1 || _fail "failed to mkfs"
SCRATCH_OPTIONS="-o fsid=erofs_${seq}.img -o device=blob.img"
_scratch_mount 2>>$seqres.full
cat $SCRATCH_MNT/testfile
_scratch_unmount

echo Silence is golden
status=0
exit 0
