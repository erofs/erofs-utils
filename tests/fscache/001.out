QA output created by 001
test
test
test
Silence is golden
